spec/jobs/sync_topic_user_bookmarked_spec.rb

```diff
@@ -66,4 +66,76 @@
     expect(tu4.reload.bookmarked).to eq(true)
     expect(tu5.reload.bookmarked).to eq(false)
   end
+
+  context "for polymorphic bookmarks" do
+    before do
+      SiteSetting.use_polymorphic_bookmarks = true
+    end
+
+    it "corrects all topic_users.bookmarked records for the topic" do
+      topic = Fabricate(:topic)
+      Fabricate(:post, topic: topic)
+      Fabricate(:post, topic: topic)
+      Fabricate(:post, topic: topic)
```

Comment:
I think we can move this to `fab!` within the context. Historically, fabrication is death by a thousand cuts so I tend to pre-optimize it.